nav2_world_model/include/nav2_world_model/region_visualizer.hpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_WORLD_MODEL__REGION_VISUALIZER_HPP_
+#define NAV2_WORLD_MODEL__REGION_VISUALIZER_HPP_
+
+#include "rclcpp/rclcpp.hpp"
+#include "visualization_msgs/msg/marker.hpp"
+
+namespace nav2_world_model
+{
+
+class RegionVisualizer
+{
+public:
+  explicit RegionVisualizer(rclcpp::Node::SharedPtr & node)
+  : node_(node),
+    marker_(visualization_msgs::msg::Marker())
+  {
+    marker_publisher_ = node_->create_publisher<visualization_msgs::msg::Marker>(
+      "world_model_cell", 1);
```

Comment:
That's probably ok for now since only one primitive is executed at a time, and the user might want to visualize only what the current primitive is requesting. We can extend this in the future as we see a need.